### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Andrey1104**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,63 @@
...
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Any:
```
Comment (Dimosphen1) [inline app/main.py:15]: Be more specific in your type annotations and indicate that `other` could be either an `int`, `float`, or `Distance` instance and that the return type would be always a `Distance` class instance. Make such changes to other methods where applicable.

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,63 @@
...
+        if isinstance(other, Distance):
+            return self.km == other.km
+        return self.km == other
+
+    def __le__(self, other: Any) -> bool:
+        if isinstance(other, Distance):
+            return self.km <= other.km
+        return self.km <= other
+
+    def __ge__(self, other: Any) -> bool:
```
Comment (Dimosphen1) [inline app/main.py:55]: You could implement some methods based on the implemented functionality of others. E.g. `self` is greater or equal to `other` when `self` is not less than `other`.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested.

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,59 @@
...
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: (int, float, Distance)) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        self.km += other
+        return self
```
Comment (viktoria-rybenchuk) [inline app/main.py:25]: write this code with one return

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,59 @@
...
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: (int, float, Distance)) -> Distance:
```
Comment (viktoria-rybenchuk) [inline app/main.py:15]:
```suggestion
    def __add__(self, other: int | float | Distance) -> Distance:
```

Comment (viktoria-rybenchuk) [reply app/main.py:15]: fix in all places

**Review by viktoria-rybenchuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**